pkg/sql/sqlbase/testutils.go

```diff
@@ -1473,6 +1551,129 @@ func randIndexTableDefFromCols(
 	return tree.IndexTableDef{Columns: indexElemList}
 }
 
+// randPartialIndexPredicateFromCols creates a partial index expression with a
+// random subset of the given columns. There is a possibility that a partial
+// index expression cannot be created for the given columns, in which case nil
+// is returned. This happens when none of the columns have types that are
+// supported for creating partial index predicate expressions. See
+// isAllowedPartialIndexColType for details on which types are supported.
+func randPartialIndexPredicateFromCols(
+	rng *rand.Rand, columnTableDefs []*tree.ColumnTableDef, tableName *tree.TableName,
+) tree.Expr {
+	// Shuffle the columns.
+	cpy := make([]*tree.ColumnTableDef, len(columnTableDefs))
+	copy(cpy, columnTableDefs)
+	rng.Shuffle(len(cpy), func(i, j int) { cpy[i], cpy[j] = cpy[j], cpy[i] })
+
+	// Select a random number of columns (at least 1). Loop through the columns
+	// to find columns with types that are currently supported for generating
+	// partial index expressions.
+	nCols := rng.Intn(len(cpy)) + 1
+	cols := make([]*tree.ColumnTableDef, 0, nCols)
+	for _, col := range cpy {
+		if isAllowedPartialIndexColType(col) {
+			cols = append(cols, col)
+		}
+		if len(cols) == nCols {
+			break
+		}
+	}
+
+	// Build a boolean expression tree with containing a reference to each
+	// column.
+	var e tree.Expr
+	for _, columnTableDef := range cols {
+		expr := randBoolColumnExpr(rng, columnTableDef, tableName)
+		// If an expression has already been built, combine the previous and
+		// current expression with an AndExpr or OrExpr.
+		if e != nil {
+			expr = randAndOrExpr(rng, e, expr)
+		}
+		e = expr
+	}
+	return e
+}
+
+// isAllowedPartialIndexColType returns true if the column type is supported and
+// the column can be included in generating random partial index predicate
+// expressions. Currently, the following types are supported:
+//
+//   - Booleans
+//   - Types that are valid in comparison operations (=, !=, <, <=, >, >=).
+//
+// This function must be kept in sync with the implementation of
+// randBoolColumnExpr.
+func isAllowedPartialIndexColType(columnTableDef *tree.ColumnTableDef) bool {
+	fam := columnTableDef.Type.(*types.T).Family()
+	switch {
+	case fam == types.BoolFamily:
```

Comment:
Can you fix this too? Multiple cases values can be separated by commas.

Reply:
Oops, yes. Thanks for pointing this out.